libvalkeycluster/valkeycluster.h

```diff
@@ -0,0 +1,358 @@
+ * the cause of the error. */
+int valkeyClusterSetConnectCallback(valkeyClusterContext *cc,
+                                    void(fn)(const valkeyContext *c,
+                                             int status));
+
+/* A hook for events. */
+int valkeyClusterSetEventCallback(valkeyClusterContext *cc,
+                                  void(fn)(const valkeyClusterContext *cc,
+                                           int event, void *privdata),
+                                  void *privdata);
+
+/* Blocking
+ * The following functions will block for a reply, or return NULL if there was
+ * an error in performing the command.
+ */
+
+/* Variadic commands (like printf) */
+void *valkeyClusterCommand(valkeyClusterContext *cc, const char *format, ...);
+void *valkeyClusterCommandToNode(valkeyClusterContext *cc,
+                                 valkeyClusterNode *node, const char *format,
+                                 ...);
+/* Variadic using va_list */
+void *valkeyClustervCommand(valkeyClusterContext *cc, const char *format,
+                            va_list ap);
+/* Using argc and argv */
+void *valkeyClusterCommandArgv(valkeyClusterContext *cc, int argc,
+                               const char **argv, const size_t *argvlen);
+/* Send a Valkey protocol encoded string */
+void *valkeyClusterFormattedCommand(valkeyClusterContext *cc, char *cmd,
+                                    int len);
+
+/* Pipelining
+ * The following functions will write a command to the output buffer.
+ * A call to `valkeyClusterGetReply()` will flush all commands in the output
+ * buffer and read until it has a reply from the first command in the buffer.
+ */
+
+/* Variadic commands (like printf) */
+int valkeyClusterAppendCommand(valkeyClusterContext *cc, const char *format,
+                               ...);
+int valkeyClusterAppendCommandToNode(valkeyClusterContext *cc,
+                                     valkeyClusterNode *node,
+                                     const char *format, ...);
+/* Variadic using va_list */
+int valkeyClustervAppendCommand(valkeyClusterContext *cc, const char *format,
+                                va_list ap);
+/* Using argc and argv */
+int valkeyClusterAppendCommandArgv(valkeyClusterContext *cc, int argc,
+                                   const char **argv, const size_t *argvlen);
+/* Use a Valkey protocol encoded string as command */
+int valkeyClusterAppendFormattedCommand(valkeyClusterContext *cc, char *cmd,
+                                        int len);
+/* Flush output buffer and return first reply */
+int valkeyClusterGetReply(valkeyClusterContext *cc, void **reply);
+
+/* Reset context after a performed pipelining */
+void valkeyClusterReset(valkeyClusterContext *cc);
+
+/* Update the slotmap by querying any node. */
+int valkeyClusterUpdateSlotmap(valkeyClusterContext *cc);
+
+/* Internal functions */
+valkeyContext *ctx_get_by_node(valkeyClusterContext *cc,
```

Comment:
Hmmm I wonder if internal functions should really be in a public header. Maybe they could/should be moved to a separate header?